### PR TITLE
Rename all inline providers with an inline:: prefix

### PR DESCRIPTION
**Pull request by ashwinb**
# What does this PR do?

Just like all remote providers are prefixed with `remote::`, we prefix Inline providers with `inline::`

Like a couple other changes pushed recently (post 0.0.50 and a few even before), this is a backward incompatible change. This means existing "images" (dockers, conda builds) won't work and you need to rebuild from source.

## Test Plan

Run `llama stack build --template meta-reference`, followed by `llama stack run meta-reference --port 5003`

### CODE REVIEW

**Comment by ashwinb** [comment]
cc @subramen @heyjustinai @init27 @jeffxtang  (we are rushing through a bunch of back-incompat changes this week now)

File: distributions/bedrock/run.yaml [hunk 1]
```diff
@@ -35,12 +35,12 @@ providers:
         region_name: <AWS_REGION>
   agents:
     - provider_id: meta0
-      provider_type: meta-reference
+      provider_type: inline::meta-reference
```
Comment (raghotham) [inline distributions/bedrock/run.yaml:38]: Is this inline::faiss?

Comment (ashwinb) [reply distributions/bedrock/run.yaml:38]: @raghotham this is the agents API so certainly not faiss. it is our _agentic loop_

Comment (raghotham) [reply distributions/bedrock/run.yaml:38]: Oops

**Review by raghotham — APPROVED** [review]

**Pull request merged**